dev-client/src/screens/LocationDashboardScreen.tsx

```diff
@@ -43,21 +46,34 @@ export const LocationDashboardScreen = ({siteId, coords}: Props) => {
     coords = site!;
   }
 
-  const appBarRightButton = useMemo(
-    () =>
-      siteId ? (
-        <AppBarIconButton
-          name="settings"
-          onPress={() => navigation.navigate('SITE_SETTINGS', {siteId})}
-        />
-      ) : (
+  const userRole = useSelector(state =>
+    siteId === undefined ? null : selectUserRoleSite(state, siteId),
+  );
+
+  const appBarRightButton = useMemo(() => {
+    // display add button if no site associated with location
+    if (!siteId) {
+      return (
         <AppBarIconButton
           name="add"
           onPress={() => navigation.navigate('CREATE_SITE', {coords})}
         />
-      ),
-    [siteId, coords, navigation],
-  );
+      );
+    }
+
+    // display nothing if user does not own the site / is not manager
```

Comment:
question: what are the possible values for `userRole`?

Reply:
`const userRole: SiteUserRole | null`

```typescript
export type SiteUserRole =
  | { kind: 'site'; role: 'owner' }
  | { kind: 'project'; role: UserRole };

export type UserRole =
  | 'contributor'
  | 'manager'
  | 'viewer';
```